Add CSS Classes to the Workflow Status Box

saifsultanc opened the pull request:
## Description
The Workflow Status Box is built with text and line breaks. This makes it difficult for customers to tap specific items on the markup for customizations (CSS rules for instance). This PR adds custom classes to make the customizations easy.

This resolves [gravityflow/backlog#100](https://github.com/gravityflow/backlog/issues/100).

Classes added:

- Each field is wrapped in a `div` of class `gravityflow-status-box-field`
- An additional class has been added to each div for specific targetting. `gravityflow-status-box-field-entry-id` for entry id, `gravityflow-status-box-field-due-date` for due date, etc.
- Each field is wrapped in span pairs of label-value. For the label, the class is `gravityflow-status-box-field-label`. For the value, the class is `gravityflow-status-box-field-value`.

## Testing instructions
1. After switching this PR, add CSS rules to different elements on the Workflow Status Box.
2. Confirm the CSS customizations works fine.
3. Repeat to check with Gravity Forms 2.4 as well.

## Automated Test Enhancements
N/A
 
## Screenshots
N/A

## Documentation Changes
N/A

## Checklist:
- [x] I've tested the code.
- [x] My code follows the WordPress code style. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/ -->
- [x] My code follows the accessibility standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code follows the inline documentation standards. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/ -->

Idealien commented on class-gravity-flow.php:
```diff
@@ -4204,7 +4204,7 @@ public function maybe_display_entry_detail_workflow_info( $current_step, $form,
 			}
 
-			printf( '%s: %s<br/><br/>', esc_html__( 'Entry ID', 'gravityflow' ), $entry_id_link );
+			printf( '<div class="gravityflow-status-box-field gravityflow-status-box-field-entry-id">%s: %s</div>', esc_html__( 'Entry ID', 'gravityflow' ), $entry_id_link );
```
One comment that could apply for each row of data output would be to ensure the label / value %s each get wrapped in a <span> with a class. WIthout, it's hard to style (or hide) each separately.

printf( '<div class="gravityflow-status-box-field gravityflow-status-box-field-entry-id"><span class="gravityflow-status-box-field-label">%s: </span><span>gravityflow-status-box-field-value">%s</span></div>', esc_html__( 'Entry ID', 'gravityflow' ), $entry_id_link );

Idealien replied:
One comment that could apply for each row of data output would be to ensure the label / value %s each get wrapped in a <span> with a class. WIthout, it's hard to style (or hide) each separately.

`printf( '<div class="gravityflow-status-box-field gravityflow-status-box-field-entry-id"><span class="gravityflow-status-box-field-label">%s: </span><span>gravityflow-status-box-field-value">%s</span></div>', esc_html__( 'Entry ID', 'gravityflow' ), $entry_id_link );`

Idealien commented on class-gravity-flow.php:
```diff
@@ -4213,41 +4213,38 @@ public function maybe_display_entry_detail_workflow_info( $current_step, $form,
 				if ( $date_created != $last_updated ) {
-					echo '<br /><br />';
-					esc_html_e( 'Last updated', 'gravityflow' ); ?>: <?php echo esc_html( $last_updated );
+					esc_html_e( '<div class="gravityflow-status-box-field gravityflow-status-box-field-last-updated">Last updated</div>', 'gravityflow' ); ?>: <?php echo esc_html( $last_updated );
```
Format through printf so that the 'Last updated' string does not include extraneous translation markup.

Idealien replied:
![image](https://user-images.githubusercontent.com/4549984/121783125-f4e48e00-cb7a-11eb-9776-a63b2e5cc6cd.png)

Idealien reviewed (changes requested):
The changes applied are great and address all of markup generated in maybe_display_entry_detail_workflow_info function. What do you think about adding similar to the maybe_display_entry_detail_step_status function? It also generates markup within the #minor-publising/.gravityflow-status-box. for the elements of 
- step name
- scheduled/expired 
- workflow detail box

![image](https://user-images.githubusercontent.com/4549984/122649876-0a6b3200-d0fe-11eb-9003-bd0013624a85.png)

Idealien commented on includes/steps/class-step-approval.php:
```diff
@@ -775,9 +775,11 @@ public function workflow_detail_box( $form, $args ) {
+					<?php printf( '<span class="gravityflow-status-box-field-label">%s </span><span class="gravityflow-status-box-field-value">(%s)</span>', $this->get_name(), $status ); ?>
+				</h4>
+			</div>
 			<div>
```
**Looking good!**

This container (of assignees), and the 'gravityflow-note', are the only two that are now missing a named container. Can it be added and check if any CSS (margins/padding) can be cleaned up since the gravityflow-status-box-field selector has a consistent amount?

Idealien reviewed (approved):
I added a few items relating to consistency between approval and user input step types. Removing some line breaks that can now better be handled via the CSS classes.

Marking PR as approved - Saif could you make 2 minor revisions before you merge please.
1) Either update change_log.txt or add to the PR description with some more details about the classes added. Will help for the release notes to highlight what people can now expect to use.

2) A small alignment difference remains on the user input step in the new branch.
![image](https://user-images.githubusercontent.com/4549984/128118708-3f4597a0-35ba-4dee-91a0-dd38b676ac56.png)

saifsultanc commented:
> I added a few items relating to consistency between approval and user input step types. Removing some line breaks that can now better be handled via the CSS classes.
> 
> Marking PR as approved - Saif could you make 2 minor revisions before you merge please.
> 
> 1. Either update change_log.txt or add to the PR description with some more details about the classes added. Will help for the release notes to highlight what people can now expect to use.
> 2. A small alignment difference remains on the assignees of user input step in the new branch.
>    ![image](https://user-images.githubusercontent.com/4549984/128118708-3f4597a0-35ba-4dee-91a0-dd38b676ac56.png)

1. Updated the PR description
2. Fixed the alignment. Both Approval and User Input step alignments match now.
